pkg/controller/remotemachineset/remotemachineset_controller_test.go

```diff
@@ -227,6 +227,100 @@ func TestRemoteMachineSetReconcile(t *testing.T) {
 				}
 			},
 		},
+		{
+			name: "Multiple machineset noop",
+			localExisting: []runtime.Object{
+				testClusterDeployment([]hivev1.MachinePool{
+					testMachinePool("alpha", 3, testName+"-worker", []string{"us-east-1a"}),
+					testMachinePool("beta", 3, testName+"-worker", []string{"us-east-1b"}),
+				}),
+				testSecret(adminKubeconfigSecret, adminKubeconfigSecretKey, testName),
+				testSecret(adminPasswordSecret, adminPasswordSecretKey, testName),
+				testSecret(sshKeySecret, sshKeySecretKey, testName),
+				testSecret(pullSecretSecret, pullSecretSecretKey, testName),
+			},
+			remoteExisting: []runtime.Object{
+				testMachineSet("foo-alpha-us-east-1a", 3, "alpha"),
+				testMachineSet("foo-beta-us-east-1b", 3, "beta"),
+			},
+			validate: func(rc client.Client, t *testing.T) {
+				rMSL, err := getRMSL(rc)
+				if err != nil {
+					t.Errorf("did not get expected remote machine sets: %v", err)
+				}
+				if len(rMSL.Items) != 2 {
+					t.Errorf("did not get expected remote machine sets: found %v", len(rMSL.Items))
+				}
+				for _, ms := range rMSL.Items {
+					if ms.Generation != 0 {
+						t.Errorf("found unexpected machine set update")
+					}
+				}
+			},
+		},
+		{
+			name: "Update multiple machineset replicas",
+			localExisting: []runtime.Object{
+				testClusterDeployment([]hivev1.MachinePool{
+					testMachinePool("alpha", 3, testName+"-worker", []string{"us-east-1a"}),
+					testMachinePool("beta", 3, testName+"-worker", []string{"us-east-1b"}),
+				}),
+				testSecret(adminKubeconfigSecret, adminKubeconfigSecretKey, testName),
+				testSecret(adminPasswordSecret, adminPasswordSecretKey, testName),
+				testSecret(sshKeySecret, sshKeySecretKey, testName),
+				testSecret(pullSecretSecret, pullSecretSecretKey, testName),
+			},
+			remoteExisting: []runtime.Object{
+				testMachineSet("foo-alpha-us-east-1a", 4, "alpha"),
+				testMachineSet("foo-beta-us-east-1b", 4, "beta"),
+			},
+			validate: func(rc client.Client, t *testing.T) {
+				rMSL, err := getRMSL(rc)
+				if err != nil {
+					t.Errorf("did not get expected remote machine sets: %v", err)
+				}
+				if len(rMSL.Items) != 2 {
+					t.Errorf("did not get expected remote machine sets: found %v", len(rMSL.Items))
+				}
+				for _, ms := range rMSL.Items {
+					if *ms.Spec.Replicas != int32(3) {
+						t.Errorf("machineset has unexpected replicas")
+					}
+				}
```

Comment:
Wondering if it would be better to replace test.validate with test.expected and list each machine set name, size, and pool expected. We're getting counts here but not necessarily zones. It would shorten up the test cases a little too.